cmake/modules/CanaryLib.cmake

```diff
@@ -1,108 +1,119 @@
-endif()
-
-# *****************************************************************************
-# Target include directories - to allow #include
-# *****************************************************************************
-target_include_directories(${PROJECT_NAME}_lib
-        PUBLIC
-        ${BOOST_DI_INCLUDE_DIRS}
-        ${CMAKE_SOURCE_DIR}/src
-        ${GMP_INCLUDE_DIRS}
-        ${LUAJIT_INCLUDE_DIRS}
-        ${PARALLEL_HASHMAP_INCLUDE_DIRS}
-        )
-
-# *****************************************************************************
-# Target links to external dependencies
-# *****************************************************************************
-target_link_libraries(${PROJECT_NAME}_lib
-    PUBLIC
-        ${GMP_LIBRARIES}
-        ${LUAJIT_LIBRARIES}
-        CURL::libcurl
-        ZLIB::ZLIB
-        absl::any absl::log absl::base absl::bits
-        asio::asio
-        eventpp::eventpp
-        fmt::fmt
-        magic_enum::magic_enum
-        mio::mio
-        protobuf::libprotobuf
-        pugixml::pugixml
-        spdlog::spdlog
-        unofficial::argon2::libargon2
-        unofficial::libmariadb
-        unofficial::mariadbclient
-)
-
-if(CMAKE_BUILD_TYPE MATCHES Debug)
-    target_link_libraries(${PROJECT_NAME}_lib PUBLIC ${ZLIB_LIBRARY_DEBUG})
-else()
-    target_link_libraries(${PROJECT_NAME}_lib PUBLIC ${ZLIB_LIBRARY_RELEASE})
-endif()
-
-if (MSVC)
-    if(BUILD_STATIC_LIBRARY)
-        target_link_libraries(${PROJECT_NAME}_lib PUBLIC jsoncpp_static)
-    else()
-        target_link_libraries(${PROJECT_NAME}_lib PUBLIC jsoncpp_lib)
-    endif()
-
-    target_link_libraries(${PROJECT_NAME}_lib PUBLIC ${CMAKE_THREAD_LIBS_INIT} ${MYSQL_CLIENT_LIBS})
-else()
-    target_link_libraries(${PROJECT_NAME}_lib PUBLIC jsoncpp_static Threads::Threads)
-endif (MSVC)
-
-# === OpenMP ===
-if(OPTIONS_ENABLE_OPENMP)
-    log_option_enabled("openmp")
-    find_package(OpenMP)
-    if(OpenMP_CXX_FOUND)
-        target_link_libraries(${PROJECT_NAME}_lib PUBLIC OpenMP::OpenMP_CXX)
-    endif()
-else()
-    log_option_disabled("openmp")
-endif()
+# Define and setup CanaryLib main library target
+add_library(${PROJECT_NAME}_lib)
+setup_target(${PROJECT_NAME}_lib)
+
+# Add subdirectories
+add_subdirectory(config)
+add_subdirectory(creatures)
+add_subdirectory(database)
+add_subdirectory(game)
+add_subdirectory(io)
+add_subdirectory(items)
+add_subdirectory(kv)
+add_subdirectory(lib)
+add_subdirectory(lua)
+add_subdirectory(map)
+add_subdirectory(security)
+add_subdirectory(server)
+add_subdirectory(utils)
+
+set(ProtobufFiles
+	protobuf/appearances.pb.cc
+	protobuf/kv.pb.cc
+)
+
+# Add more global sources - please add preferably in the sub_directory CMakeLists.
+target_sources(${PROJECT_NAME}_lib PRIVATE canary_server.cpp ${ProtobufFiles})
+
+# Skip unity build inclusion for protobuf files
+set_source_files_properties(
+    ${ProtobufFiles} PROPERTIES SKIP_UNITY_BUILD_INCLUSION ON
+)
```

Comment:
Maybe those should  go to protobuf/CMakeFiles/txt?

Reply:
I tried that, but my cmake foo ran out. That directory is in its own "project" so things were breaking in confusing ways for me.